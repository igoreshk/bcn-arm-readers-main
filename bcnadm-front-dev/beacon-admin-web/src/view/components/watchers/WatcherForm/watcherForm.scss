.visitorForm {
  .firstBlock {
    display: flex;
    justify-content: space-between;
    align-items: center;

    @media screen and (max-width: 650px) {
      flex-direction: column;
    }

    .dropzone {
      min-width: 250px;
      height: 200px;
      border: dashed 1px;
      border-color: #999999;
      cursor: pointer;
      font-size: 13px;
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      text-align: center;
      margin-left: 20px;

      @media screen and (max-width: 650px) {
        min-width: unset;
        width: 100%;
        margin-left: 0;
        margin-top: 20px;
      }
    }
    
    .preview {
      width: 100%;
      height: 100%;
    }

    .error {
      color: #FF0000;
      margin-top: 20px;
    }
  }

  .secondBlock {
    .minorAndMajorFields {
      display: flex;

      @media screen and (max-width: 425px) {
        flex-direction: column;
      }

      > div:first-child {
        margin-right: 15px;

        @media screen and (max-width: 425px) {
          margin-right: 0;
        }
      }
    }
  }
}