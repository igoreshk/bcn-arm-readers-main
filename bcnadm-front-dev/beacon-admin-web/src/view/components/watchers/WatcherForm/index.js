import VisitorsSelectField from 'src/view/components/visitors/VisitorsSelectField';
import React from 'react';
import { Dialog, DialogTitle, DialogContent, DialogActions, Button } from '@material-ui/core';
import PropTypes from 'prop-types';
import { Form, Field } from 'react-final-form';
import WatcherFormTextField from '../WatcherFormTextField';
import './watcherForm.scss';

const WatcherForm = (props) => {
  const {
    handleOnChange,
    watcher,
    translate,
    cancel,
    selectVisitorsIds,
    visitorsList,
    saveWatcher,
    isLoading,
    isAppropriateName
  } = props;

  const onClickSubmit = () => {
    let isValid;
    if (watcher.name && watcher.name.length && isAppropriateName) {
      isValid = true;
    } else {
      isValid = false;
    }
    return isValid ? saveWatcher() : '';
  };

  const onSubmit = (event) => {
    event.preventDefault();
    onClickSubmit();
  };

  return (
    <Form
      onSubmit={onSubmit}
      render={() => (
        <Dialog open onClose={cancel}>
          <DialogTitle className="watcherDialogTitle">
            {watcher.entityId ? props.translate('watchers.editWatcher') : props.translate('watchers.newWatcher')}
          </DialogTitle>
          <DialogContent>
            <form className="visitorForm">
              <Field>
                {() => (
                  <div className="firstBlock">
                    <WatcherFormTextField
                      name="watcher-name"
                      hintText={props.translate('monitoring.name')}
                      floatingLabelText={props.translate('monitoring.name')}
                      input={watcher.name ? watcher.name : ''}
                      onChange={handleOnChange}
                      isAppropriateName={isAppropriateName}
                      helperText={props.translate('watchers.nonUniqueWatcherName')}
                    />
                  </div>
                )}
              </Field>
              <Field>
                {() => (
                  <div className="secondGroup">
                    <VisitorsSelectField
                      visitors={visitorsList}
                      selectVisitorsIds={selectVisitorsIds}
                      translate={translate}
                      watcher={watcher}
                      isLoading={isLoading}
                    />
                  </div>
                )}
              </Field>
            </form>
          </DialogContent>
          <DialogActions>
            <Button disabled={isLoading} onClick={cancel}>
              {props.translate('visitors.dialog.cancel')}
            </Button>
            <Button
              disabled={!(watcher.name && watcher.name.length && isAppropriateName) || isLoading}
              type={'submit'}
              onClick={onSubmit}
              color="primary"
            >
              {watcher.entityId ? props.translate('visitors.dialog.save') : props.translate('visitors.dialog.create')}
            </Button>
          </DialogActions>
        </Dialog>
      )}
    />
  );
};

WatcherForm.propTypes = {
  handleOnChange: PropTypes.func.isRequired,
  cancel: PropTypes.func.isRequired,
  visitorsList: PropTypes.arrayOf(PropTypes.object),
  watcher: PropTypes.object,
  translate: PropTypes.func.isRequired,
  selectVisitorsIds: PropTypes.func.isRequired,
  saveWatcher: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired
};

export default WatcherForm;
