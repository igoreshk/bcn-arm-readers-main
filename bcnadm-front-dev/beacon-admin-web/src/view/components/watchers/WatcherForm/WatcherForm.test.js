import React from 'react';
import { shallow, mount } from 'enzyme';
import { Button } from '@material-ui/core';
import WatcherForm from './index';

const props = {
  handleOnChange: jest.fn(),
  cancel: jest.fn(),
  visitorsList: [],
  isLoading: false,
  watcher: {
    entityId: 'someId',
    name: 'someName',
    visitorIds: []
  },
  translate: () => '',
  selectVisitorsIds: jest.fn(),
  saveWatcher: jest.fn(),
  isAppropriateName: null
};

describe('WatcherForm', () => {
  const shallowWrapper = shallow(<WatcherForm {...props} />);
  const wrapper = mount(<WatcherForm {...props} />);

  it('Check matching snapshot', () => expect(shallowWrapper).toMatchSnapshot());

  describe('Form header', () => {
    it('should render correct header if the watcher is new', () => {
      wrapper.setProps({
        translate: () => 'Add watcher'
      });

      const title = wrapper.find('.watcherDialogTitle').first();
      expect(title.text()).toEqual('Add watcher');
    });

    it('should render correct header if the watcher is edited', () => {
      wrapper.setProps({
        watcher: {
          entityId: 1,
          name: 'name'
        },
        translate: () => 'Edit watcher'
      });
      const title = wrapper.find('.watcherDialogTitle').first();
      expect(title.text()).toEqual('Edit watcher');
    });
  });

  describe('testSaveButton', () => {
    let saveButton = wrapper.find(Button).at(1);

    it('Check initial state of button is disabled ', () => {
      wrapper.setProps({
        watcher: {
          entityId: 1
        }
      });
      saveButton = wrapper.find(Button).at(1);
      expect(saveButton.prop('disabled')).toEqual(true);
    });

    it('Check button is enabled after name was changed and it is appropriate', () => {
      wrapper.setProps({
        watcher: {
          entityId: 1,
          name: 'name'
        },
        isAppropriateName: true
      });
      saveButton = wrapper.find(Button).at(1);
      expect(saveButton.prop('disabled')).toEqual(false);
    });

    it('Check button is disabled after name was changed and it is not appropriate', () => {
      wrapper.setProps({
        watcher: {
          entityId: 1,
          name: 'name'
        },
        isAppropriateName: false
      });
      saveButton = wrapper.find(Button).at(1);
      expect(saveButton.prop('disabled')).toEqual(true);
    });

    it('renders Save text if the watcher is edited', () => {
      wrapper.setProps({
        watcher: {
          entityId: 1,
          name: 'name'
        },
        translate: () => 'Save'
      });
      saveButton = wrapper.find(Button).at(1);
      expect(saveButton.text()).toEqual('Save');
    });
  });
});
