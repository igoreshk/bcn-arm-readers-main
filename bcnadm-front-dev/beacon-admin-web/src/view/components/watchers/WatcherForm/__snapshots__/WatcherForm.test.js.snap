// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WatcherForm Check matching snapshot 1`] = `
<ReactFinalForm
  onSubmit={[Function]}
  render={[Function]}
/>
`;
